src/Redux/actionCreators/ServiceActions.js

```diff
@@ -29,21 +31,25 @@ export const fetchService = pagination => async dispatch => {
 
 export const invokeServiceMethod = data => dispatch => {
   dispatch(loaderActions.startAppLoader(LoaderContent.SERVICE_INVOKATION));
-  Auth.currentSession({ bypassCache: true }).then(currentSession => {
-    const apiName = APIEndpoints.INVOKE_SERVICE.name;
-    const path = `${APIPaths.INVOKE_SERVICE}-${data.service_id}`;
-    let myInit = {
-      body: data,
-      headers: { Authorization: currentSession.idToken.jwtToken },
-    };
-    API.post(apiName, path, myInit).then(response => {
-      dispatch(loaderActions.stopAppLoader);
-      dispatch({
-        type: UPDATE_SERVICE_EXECUTION_RESPONSE,
-        payload: { response: JSON.parse(response.data), isComplete: true },
-      });
-    });
-  });
+  Auth.currentSession({ bypassCache: true })
+    .then(currentSession => {
+      const apiName = APIEndpoints.INVOKE_SERVICE.name;
+      const path = APIPaths.INVOKE_SERVICE;
+      let myInit = {
+        body: data,
+        headers: { Authorization: currentSession.idToken.jwtToken },
+      };
+      API.post(apiName, path, myInit)
+        .then(response => {
```

Comment:
You could chain the promises rather than nesting them.

Reply:
[done](https://github.com/singnet/refactored-dapp/pull/35/commits/16e375d320914819b20dddec43d4170b76d66144)